internal/pkg/agent/application/coordinator/coordinator.go

```diff
@@ -825,3 +818,24 @@ func hasState(components []runtime.ComponentComponentState, state client.UnitSta
 	}
 	return false
 }
+
+func logBasedOnState(l *logger.Logger, state client.UnitState, msg string, args ...interface{}) {
+	switch state {
+	case client.UnitStateStarting:
+		l.With(args...).Info(msg)
+	case client.UnitStateConfiguring:
+		l.With(args...).Info(msg)
+	case client.UnitStateDegraded:
+		l.With(args...).Warn(msg)
+	case client.UnitStateHealthy:
+		l.With(args...).Info(msg)
+	case client.UnitStateFailed:
+		l.With(args...).Error(msg)
+	case client.UnitStateStopping:
+		l.With(args...).Info(msg)
+	case client.UnitStateStopped:
+		l.With(args...).Info(msg)
+    default:
+        l.With(args...).Info(msg)
```

Comment:
```suggestion
	default:
		l.With(args...).Info(msg)
```

Need to manually paste in tabs when editing this remotely...